python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -66,6 +66,7 @@ class TimeWindowPartitionsDefinition(
         [
             ("start", PublicAttr[datetime]),
             ("timezone", PublicAttr[str]),
+            ("end", PublicAttr[datetime]),
```

Comment:
```suggestion
            ("end", PublicAttr[Optional[datetime]]),
```

should fix the type error

Reply:
I'm running `make pyright` locally. But it takes very long time. Is that normal?

Reply:
It takes a very long time the first time. After that, I believe, it should take about a minute.